### PR TITLE
adding getparams function to read from __bx_creds

### PR DESCRIPTION
**Pull request by bvennam**

### CODE REVIEW

**Comment by bvennam** [comment]
@anweshan This is what I was thinking for the getParams function.

My guess is that the service name can be generated -- it's just the cloud foundry service name.  To determine what the fields would be for conversation, I just did a `bx service bind` to see what was available in the __bx_creds.  

Let me know if you had any questions or concerns about this.

File: actions/conversation-v1/create-counterexample.js [hunk 1]
```diff
@@ -64,13 +64,18 @@ module.exports.test = main;
...
-    params.password = theParams.password || theParams.__bx_creds[theServiceName].password;
-    return params;
   }
+  console.log(theParams)
+  if(theParams.__bx_creds) {
+    bx_creds = theParams.__bx_creds[theServiceName]
+    delete theParams.__bx_creds
+  }
+  const params = theParams;
+  params.url = theParams.url || bx_creds.url;
```
Comment (germanattanasio) [inline actions/conversation-v1/create-counterexample.js:74]: why don't we pull all the values from `bx_creds` and let the SDK decide what to use?

File: actions/conversation-v1/create-counterexample.js [hunk 2]
```diff
@@ -64,13 +64,18 @@ module.exports.test = main;
 function getParams(theParams, theServiceName) {
   if (Object.getOwnPropertyNames(theParams).length === 0) {
```
Comment (germanattanasio) [inline actions/conversation-v1/create-counterexample.js]:
```
(Object(theParams).keys().length === 0)
```

File: actions/conversation-v1/create-counterexample.js [hunk 3]
```diff
@@ -64,13 +64,18 @@ module.exports.test = main;
...
     return theParams;
-  } else {
```
Comment (germanattanasio) [inline actions/conversation-v1/create-counterexample.js]: Something like
```js
const params = Object.assign({}, theParams.__bx_creds[theServiceName], theParams);
delete params.__bx_creds;
```

Should work

**Review by germanattanasio — CHANGES_REQUESTED** [review]

File: actions/conversation-v1/create-counterexample.js (hunk 2 above)
Comment (bvennam) [reply actions/conversation-v1/create-counterexample.js]: this results in an error for me -- i think it should be:  
```Object.keys(theParams).length === 0```

File: actions/conversation-v1/create-counterexample.js (hunk 3 above)
Comment (bvennam) [reply actions/conversation-v1/create-counterexample.js]: Just tried this, works for me & is much cleaner.  Thanks @germanattanasio

**Review by anweshan — APPROVED** [review]

**Pull request closed without merging**